packages/cli/src/utils/get-item-count-by-connection-type.ts

```diff
@@ -0,0 +1,15 @@
+import type { ITaskData } from 'n8n-workflow';
+
+export function getItemCountByConnectionType(data: ITaskData['data']): Record<string, number[]> {
```

Comment:
Shouldn't the return type be `Record<NodeConnectionType, number[]>`? I know the type of `ITaskDataConnections` accepts any string key, but in the push message we have defined the type to be a record with keys of `NodeConnectionType`, so we need to do the type narrowing somewhere, and I think this function is a good place. I guess we can just cast it

Reply:
Good point. Updated and added node connection type typeguard.